packages/did-ipfs-store/src/didStore.ts

```diff
@@ -27,9 +27,22 @@ export class DidStore implements IDidStore {
     private headers: Record<string, string> = {}
   ) {}
 
-  async save(claim: string): Promise<string> {
+  /**
+   * @param claim stringified claim. Supported types of claim content are `string` and `object`
+   * @param pinTimeout Defines how to long to wait for pinning completion before throwing error, seconds
+   */
+  async save(claim: string, pinTimeout = 5): Promise<string> {
     const blob = new Blob([claim]);
     const { cid } = await this.add(blob);
+
+    const { default: waitFor } = await (eval('import(`p-wait-for`)') as Promise<
+      typeof import('p-wait-for')
+    >);
+
+    await waitFor(async () => await this.isPinned(cid), {
+      timeout: { milliseconds: pinTimeout * 1000 },
+    });
+
```

Comment:
@JGiter could you explain a little more why this is necessary please? 

From the IPFS Cluster documentation, it is a bit unclear to me what the consistency semantics of the `add` operation are and how that aligns with the consistency semantics that this `did-store` `save` operation promises.
The [IPFS Cluster](https://ipfscluster.io/documentation/guides/pinning/#adding-files) documentation says:
> This means that when the add process is finished, your file will have been fully added to several IPFS daemons (and not necessarily the local one). 

So to me, this suggests that when `add` returns, one can be sure that IPFS Cluster has received the data (and is propagating it).

Maybe it is better to therefore have this `save` method return when the `add` method returns? In this way, the caller of `save` method could call it again before needing to wait for the full propagation to each node to be complete? 

Maybe as you have it is better, but just trying to understanding the rational a bit better 😃

Reply:
This delay was added in orded to ensure the behavior you quoted from IPFS Cluster documentation. When I tested DidStore locally data were available immediately after `add` returns, but during testing in Github Action pin status after `add` was `pinning`. May be this is because implementation of `nftstorage/ipfs-cluster` on which our implementation is based, is work in progress now

Reply:
@JGiter yes, makes sense. In principle, could we return while the status was still `pinning`? I'm not saying that's better necessarily, but just curious what the pros/cons are in your mind.

Reply:
I don't see pros of returning when data is still pinning. User will still need to wait for the data to become available. Thus task of awaiting will fall on him

Reply:
> User will still need to wait for the data to become available. Thus task of awaiting will fall on him

They might not need to await for the data to be available if they aren't trying to read the data. If they are just writing, then they don't need to wait for read availability.

> Thus task of awaiting will fall on him

I think this could actually be an advantage, as `waitFor`'s behaviour might not suit every client. Currently, there is a `pinTimeout` command, but what if clients want to control other params of the `waitFor` behaviour (e.g. polling frequency)

How about this as a solution? We keep the `save` method as is, but make the `add` method public so that a client could use it can have more control over the wait behaviour if they want to?

Reply:
The question is what is default behaviour? To wait or not to wait? I beleive that by default we should wait. The drawback is that those who doesn't need after save availability has to provide empty `waitForPinned`. But the drowback for the case when default `waitForPinned` is empty is bigger, because whoever need claim to be avalilable right after `save` will need to put significant efforts for `waitForPinned` implementation. Also it might be confusing for people that `save` doesn't saves